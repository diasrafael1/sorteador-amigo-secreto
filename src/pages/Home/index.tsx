import Card from "../../components/Card";
import Footer from "../../components/Footer";
import Form from "../../components/Form";
import Header from "../../components/Header";
import ListPeoples from "../../components/ListPeoples";

export default function Home() {
  return (
    <>
      <Header />
      <Card>
        <h1>Vamos come√ßar!</h1>
        <Form />
        <ListPeoples />
        <Footer />
      </Card>
    </>
  );
}
