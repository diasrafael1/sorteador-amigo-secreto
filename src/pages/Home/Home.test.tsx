import { render } from "@testing-library/react";
import { RecoilRoot } from "recoil";
import Home from ".";

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => {
  return {
    useNavigate: () => mockNavigate,
  };
});

describe("a pagina de configuração", () => {
  test("deve ser renderizada corretamente", () => {
    //Teste de snapshot
    const { container } = render(
      <RecoilRoot>
        <Home />
      </RecoilRoot>
    );

    //irá tirar um snapshot do componente e salvará o html na pasta __snapshot__
    //se o componente estiver diferente de como está salvo no __snapshot__ o teste dará erro.
    expect(container).toMatchSnapshot();
  });
});
