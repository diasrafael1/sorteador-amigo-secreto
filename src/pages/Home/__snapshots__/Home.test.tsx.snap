// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a pagina de configuração deve ser renderizada corretamente 1`] = `
<div>
  <header
    class="sc-eDvSVe bgFaSt"
  >
    <img
      alt="Logo"
      class="logoSmall"
      src="logo-pequeno.png"
    />
    <img
      alt="Logo"
      class="logo"
      src="logo.png"
    />
    <img
      alt="Pessoa"
      class="person"
      src="participante.png"
    />
  </header>
  <section
    class="sc-bcXHqe kaIdmH"
  >
    <h1>
      Vamos começar!
    </h1>
    <form
      class="sc-hLBbgP kURBDR"
    >
      <input
        placeholder="Insira os nomes do Participantes"
        type="text"
        value=""
      />
      <button
        class="sc-dkrFOg ceqLqg"
        disabled=""
      >
        Adicionar
      </button>
    </form>
    
    <ul
      class="sc-jSUZER bAkzxc"
    />
    <footer
      class="sc-gswNZR bnPSAZ"
    >
      <button
        class="sc-dkrFOg ceqLqg"
        disabled=""
      >
        Iniciar Brincadeira!
      </button>
      <img
        alt="Imagem de sacolas"
        src="sacolas.png"
      />
    </footer>
  </section>
</div>
`;
